### ISSUE TITLE
Add explanation of what are schema, uiSchema and displayFields

### ISSUE THREAD

**Issue by enguerran** [Kinto/kinto-admin#109]
I saw the list of records as below:

![previousfieldnames](https://cloud.githubusercontent.com/assets/701648/15703357/15b66048-27e5-11e6-84c5-094e748494d3.png)

Then I go to the collection properties edition to change de fieldnames as below:

![rename2](https://cloud.githubusercontent.com/assets/701648/15703435/7e525fda-27e5-11e6-8a61-813b816e05a0.png)

Then I go back to the list of records, every items are there but fields are not displaying the correct values. Or I may have a wrong understanding of what is the collection properties displayFields is.

![newfieldname](https://cloud.githubusercontent.com/assets/701648/15703410/606d7c2a-27e5-11e6-8f1f-8889655aa257.png)

**Comment by n1k0** [comment]
Yes that's expected, as "citation" and "auteur" are not members of the defined JSON schema. How would you think we could make that clearer?

**Comment by Natim** [comment]
@enguerran Were you expecting that the records fields name go renamed for all records?

**Comment by enguerran** [comment]
I though _displayFields_ was to rename the fields. (or delete or create a new one)
![screenshot from 2016-06-01 13-30-35](https://cloud.githubusercontent.com/assets/701648/15708089/d957b67a-27fc-11e6-8b2f-d0d2e8320a90.png)

**Comment by Natim** [comment]
No it is just the name of the column you want on the collection list.

**Comment by enguerran** [comment]
How it is linked to the schema?
So if I wanted to rename the column on the collection list, i can't?

**Comment by Natim** [comment]
Oh you mean we should use the title from the schema to rename it column?

**Comment by enguerran** [comment]
I though it was that.
Since it is not, I cannot understand what this feature is about: why is it for?
I mean, the text input is ok and very clear when you _add_ a new field, but what does it do when you _edit_ the fieldName?

**Comment by n1k0** [comment]
> I though _displayFields_ was to rename the fields. (or delete or create a new one)
> ![screenshot from 2016-06-01 13-30-35](https://cloud.githubusercontent.com/assets/701648/15708089/d957b67a-27fc-11e6-8b2f-d0d2e8320a90.png)

Actually the entries for displayFields should map to the schema object property keys, here *field1* and *field2*. If you want to tweak the way the field names are rendered, you should use the `title` json schema property in the schema field definitions.

Okay that's complicated. So how should we makes this clear?

**Comment by enguerran** [comment]
Okay, I may be dumb, but even when changing the whole schema, I cannot change the property name nor the column title and keep the data. Is that a correct behavior or what?

![screen shot 2016-06-01 at 14 10 22](https://cloud.githubusercontent.com/assets/701648/15709046/c53d987a-2802-11e6-9188-1322bd06f0da.png)

**Comment by Natim** [comment]
@enguerran You can create the JSON schema using the formbuilder and then press Show JSON: http://kinto.github.io/formbuilder/#/builder

You are looking at something like:
```
  "properties": {
    "citation": {
      "type": "string",
      "title": "Citation",
      "description": "",
    }
```

**Comment by n1k0** [comment]
> Okay, I may be dumb, but even when changing the whole schema, I cannot change the property name nor the column title and keep the data. Is that a correct behavior or what?

Yes, there's no way to migrate your data automatically, and I'm not sure - even if that's appealing - we're gonna support this anytime soon :)

Sure people may be used to renamed columns in mysql or RDMS but we're talking nosql here, and I'm not aware of any nosql db that allows that. Most of them I've studied use migration scripts, which in your case is probably what you're supposed to do :)

**Comment by Natim** [comment]
@n1k0 apparently that's not about renaming fields, that's about renaming the column title.

**Comment by enguerran** [comment]
@Natim yes, that right. I though it was about renaming column title. But I was wrong, so it is a little bit confusing.

@n1k0 so if we cannot rename the field in the DB, why is there an editable input field? That is what I cannot understand.

**Comment by Natim** [comment]
@enguerran Because kinto is schema less and that this schema is just the kinto-admin settings to let you better handle the data.

**Comment by n1k0** [comment]
> why is there an editable input field?

I'm sorry that's confusing, which editable input field are you referring to?

**Comment by enguerran** [comment]
Those one

![screenshot from 2016-06-01 13-30-35](https://cloud.githubusercontent.com/assets/701648/15709574/c1775674-2805-11e6-83bf-1e1b8c61b3f1.png)

**Comment by enguerran** [comment]
But I succeed to do what I wanted to do at the first place:

![screen shot 2016-06-01 at 14 33 39](https://cloud.githubusercontent.com/assets/701648/15709595/db26f598-2805-11e6-9203-0066ec6e91af.png)

giving that

![screen shot 2016-06-01 at 14 33 33](https://cloud.githubusercontent.com/assets/701648/15709600/e25725d6-2805-11e6-947e-14c1a27d8c64.png)

**Comment by Natim** [comment]
Note that you should start fresh and rename ``field1`` and ``field2`` with ``citation`` and ``author`` directly.

**Comment by n1k0** [comment]
Guys, I'm having a hard time considering this an actual bug. How should we qualify the next actionable steps to solve this? Or should we just close this issue now things are clearer?

**Comment by enguerran** [comment]
You can just close this issue.
If there is a doc somewhere, I did not read it.
If not, we should open a new issue/feature: write something about the interface or the creation process.

**Comment by Natim** [comment]
After having discuss that on IRC, I think for now the best and simplest course of action is to fix #117 and add a paragraph of text telling this:

```
1. First create a JSONSchema which describe your collection format
2. Define some uiSchema if needed to change the field widget in the admin.
3. Liste the field name you want in the column of the collection record list.
```
